### ISSUE TITLE
Example combining wasm + simplejit

### ISSUE THREAD

**Issue by kazimuth** [CraneStation/cranelift#447]
I'd like to use CraneLift as a JIT backend for a standalone WebAssembly embedding. However, from browsing the documentation I'm not sure how to do this, or even if it's fully possible yet. An example showing how to use cranelift_wasm and cranelift_simplejit together would be really helpful.

**Comment by lachlansneff** [comment]
Nebulet doesn't use the simplejit abstraction, but it does have a Webassembly implementation that uses Cranelift. Here's the bulk of the code: https://github.com/nebulet/nebulet/blob/master/src/wasm/mod.rs.

**Comment by MSleepyPanda** [comment]
I think there is https://github.com/CraneStation/wasmtime, but i don't know how well it works in the wild

**Comment by sunfishcode** [comment]
Yes; wasmtime is aiming to be a standalone WebAssembly engine useful for embedding. It's currently in development, but it can run simple programs today. I'm working on finishing the remaining features needed for wasmtime to be generally usable, though I've also been busy with a lot of other projects.

The cranelift-codegen parts needed to run wasm are pretty stable; they're in use in some larger projects (including Nebulet, mentioned above!). The main work remaining for wasmtime to be usable is in things like module instantiation, connecting imports to exports, and indirect call handling. If anyone is interested in helping, I'd be happy to mentor.
